core/src/main/java/com/netflix/conductor/core/orchestration/ExecutionDAOFacade.java

```diff
@@ -215,21 +215,19 @@ public String updateWorkflow(Workflow workflow) {
             workflow.setEndTime(System.currentTimeMillis());
         }
         executionDAO.updateWorkflow(workflow);
-        if (workflow.getStatus().isTerminal()) {
-            if (config.enableAsyncIndexing()) {
-                if (workflow.getEndTime() - workflow.getStartTime() < config.getAsyncUpdateShortRunningWorkflowDuration() * 1000) {
-                    final String workflowId = workflow.getWorkflowId();
-                    DelayWorkflowUpdate delayWorkflowUpdate = new DelayWorkflowUpdate(workflowId);
-                    LOGGER.debug("Delayed updating workflow: {} in the index by {} seconds", workflowId, config.getAsyncUpdateDelay());
-                    scheduledThreadPoolExecutor.schedule(delayWorkflowUpdate, config.getAsyncUpdateDelay(), TimeUnit.SECONDS);
-                    Monitors.recordWorkerQueueSize("delayQueue", scheduledThreadPoolExecutor.getQueue().size());
-                } else {
-                    indexDAO.asyncIndexWorkflow(workflow);
-                }
-                workflow.getTasks().forEach(indexDAO::asyncIndexTask);
+        if (config.enableAsyncIndexing()) {
+            if (workflow.getStatus().isTerminal() && workflow.getEndTime() - workflow.getStartTime() < config.getAsyncUpdateShortRunningWorkflowDuration() * 1000) {
+                final String workflowId = workflow.getWorkflowId();
+                DelayWorkflowUpdate delayWorkflowUpdate = new DelayWorkflowUpdate(workflowId);
+                LOGGER.debug("Delayed updating workflow: {} in the index by {} seconds", workflowId, config.getAsyncUpdateDelay());
+                scheduledThreadPoolExecutor.schedule(delayWorkflowUpdate, config.getAsyncUpdateDelay(), TimeUnit.SECONDS);
+                Monitors.recordWorkerQueueSize("delayQueue", scheduledThreadPoolExecutor.getQueue().size());
             } else {
-                indexDAO.indexWorkflow(workflow);
+                indexDAO.asyncIndexWorkflow(workflow);
             }
+            workflow.getTasks().forEach(indexDAO::asyncIndexTask);
```

Comment:
As part of async indexing, the contract is that task data for a workflow will only be indexed when a workflow reaches a terminal state. Please move this line to within the if block above.

Reply:
Oh, now I understand what the intent was...thanks for that. I think I covered that now. I took out the isTerminal() check earlier so I added it later.